[#960] fix(dashboard): simplify dependency and correct the startup script

yl09099 opened the pull request:
### What changes were proposed in this pull request?

simplify dependency and correct the startup script.

### Why are the changes needed?

(Please clarify why the changes are needed. For instance,

Fix: #960 

### Does this PR introduce _any_ user-facing change?

(Please list the user-facing changes introduced by your change, including

No.

### How was this patch tested?

Needn't.

yl09099 commented:
> What's this pr the relationship of #1326

Same PR. I'm changing it after the merger.

yl09099 commented:
> #1326

Some fixes based on PR #1326.I'm making changes after the merger.

yl09099 commented:
@zuston @xianjingfeng @jerqi  Modify the dashboard pom dependencies, modify the startup script to be independent of Hadoop, and separate the dashboard configuration file from the Coordinator configuration file so that it can be deployed independently.

zuston reviewed (dismissed):
LGTM.

xianjingfeng commented:
Could you fix these two problems in the pr? @yl09099 
https://github.com/apache/incubator-uniffle/pull/1339#issuecomment-1831538825

yl09099 commented:
> Could you fix these two problems in this pr? @yl09099 [#1339 (comment)](https://github.com/apache/incubator-uniffle/pull/1339#issuecomment-1831538825)

This has been modified.

xianjingfeng commented:
> > Could you fix these two problems in this pr? @yl09099 [#1339 (comment)](https://github.com/apache/incubator-uniffle/pull/1339#issuecomment-1831538825)
> 
> This has been modified.

It seems that one of them was be fixed. But we still must set the env `HADOOP_HOME`, right?

yl09099 commented:
> > > Could you fix these two problems in this pr? @yl09099 [#1339 (comment)](https://github.com/apache/incubator-uniffle/pull/1339#issuecomment-1831538825)
> > 
> > 
> > This has been modified.
> 
> It seems that one of them was fixed. But we still must set the env `HADOOP_HOME`, right?

I have deleted the configurations related to start-dashboard and HDFS, and do not need to set HADOOP_HOME on the dashboard.

yl09099 commented:
> > > Could you fix these two problems in this pr? @yl09099 [#1339 (comment)](https://github.com/apache/incubator-uniffle/pull/1339#issuecomment-1831538825)
> > 
> > 
> > This has been modified.
> 
> It seems that one of them was fixed. But we still must set the env `HADOOP_HOME`, right?

I have deleted the HDFS configuration in start-dashboard and do not need to set HADOOP_HOME on dashboard.

xianjingfeng commented:
> > > > Could you fix these two problems in this pr? @yl09099 [#1339 (comment)](https://github.com/apache/incubator-uniffle/pull/1339#issuecomment-1831538825)
> > > 
> > > 
> > > This has been modified.
> > 
> > 
> > It seems that one of them was fixed. But we still must set the env `HADOOP_HOME`, right?
> 
> I have deleted the HDFS configuration in start-dashboard and do not need to set HADOOP_HOME on dashboard.


Is there anything problems with the following code?
https://github.com/apache/incubator-uniffle/blob/ecbf2e79423505dd44bf54c612572c5c9da71175/bin/utils.sh#L175-L178

yl09099 commented:
> > > > > Could you fix these two problems in this pr? @yl09099 [#1339 (comment)](https://github.com/apache/incubator-uniffle/pull/1339#issuecomment-1831538825)
> > > > 
> > > > 
> > > > This has been modified.
> > > 
> > > 
> > > It seems that one of them was fixed. But we still must set the env `HADOOP_HOME`, right?
> > 
> > 
> > I have deleted the HDFS configuration in start-dashboard and do not need to set HADOOP_HOME on dashboard.
> 
> Is there anything problems with the following code?
> 
> https://github.com/apache/incubator-uniffle/blob/ecbf2e79423505dd44bf54c612572c5c9da71175/bin/utils.sh#L175-L178

Learned, changed

xianjingfeng commented on bin/utils.sh:
```diff
@@ -172,7 +177,9 @@ function load_rss_env {
     exit 1
   fi
-  if [ -z "$HADOOP_HOME" ]; then
+  if [[ $is_dashboard -eq 0 || -z "$HADOOP_HOME" ]]; then
+    echo "Dashboard need not HADOOP_HOME."
+  elif [[ -z "$HADOOP_HOME" ]]; then
     echo "No env HADOOP_HOME."
     exit 1
   fi
```
```suggestion
  if [[ -z "$HADOOP_HOME" ]]; then
    if [[ $is_dashboard -eq 0 ]]; then
      echo "Dashboard need not HADOOP_HOME."
    else
      echo "No env HADOOP_HOME."
      exit 1
    fi
  fi
```

xianjingfeng commented on bin/utils.sh:
```diff
@@ -152,6 +152,11 @@ function is_port_in_use {
 function load_rss_env {
   set -o allexport
 
+  is_dashboard=1
+  if [ $# -eq 1 ]; then
+          is_dashboard=$1
+  fi
```
```suggestion
  is_dashboard=0
  if [ $# -eq 1 ]; then
    is_dashboard=1
  fi
```

xianjingfeng replied:
@zuston @jerqi Do you have any idea about this part?

xianjingfeng reviewed (approved):
LGTM

zuston commented:
Thanks @yl09099 @xianjingfeng
